apps/nuxt3-ssr/components/CatalogueCard.vue

```diff
@@ -0,0 +1,131 @@
+<script setup lang="ts">
+const route = useRoute();
+import { INetwork } from "~/interfaces/types";
+import { computed } from "vue";
+
+const props = withDefaults(
+  defineProps<{
+    network: INetwork;
+    schema: string;
+    compact?: boolean;
+  }>(),
+  {
+    compact: false,
+  }
+);
+
+const catalogue = route.params.catalogue || "all";
+
+const articleClasses = computed(() => {
+  return props.compact ? "py-5 lg:px-12.5 p-5" : "lg:px-12.5 py-12.5 px-5";
+});
+
+const subtitleClasses = computed(() => {
+  return props.compact ? "hidden md:block" : "mt-1.5 block md:inline";
+});
+
+const titleContainerClasses = computed(() => {
+  return props.compact ? "flex items-center" : "";
+});
+
+const headerClasses = computed(() => {
+  return props.compact ? "" : "items-start xl:items-center";
+});
+
+const iconStarClasses = computed(() => {
+  return props.compact ? "" : "items-baseline xl:items-center mt-0.5 xl:mt-0";
+});
```

Comment:
code duplication why not use slots

Reply:
can you elaborate? I would expect somehow to make components that do this for us somehow?

Reply:
something like `
<Card >
<slot header>
<slot image>
<slot description>
<slot links> 
<Card>`

  maybe with some config props. Or instead of slots a few predefined card types to choose from ( CardWithImage, CardWithSections , ...) that have props ( or combination of slots and props ) we can then wrap these with catalogue specific components <CohortCard><CardWithSections/> <CohortCard> ( in this case the cohort cards takes a number of cohort specific items and places them / passes them to the 'CardWithSections'. The CardWithSections ( typed card) or Cards with lots of optional slots can then handle all the little class setting stuff.

Reply:
So I am trying to not redesign the components in this PR so I simply copied the existing pages. So do you agree if we accept the duplication for now?

I also believe that with som effort we can dramatically simplify the card creation (I would ideally have some configuration json object that is ideally an extension of emx2 so that in simple cases I just need to paste some emx2 columns, probably based on the generic cards you already made).